### PR TITLE
Add deviation(s) by providing a date instead of minutes

### PR DESCRIPTION
**Pull request by piehei**
This PR amends the deviations adding mechanism so that the teacher can give a date instead of pre-calculated minutes.

There's a new field in the form, error handling for that, and a new method in the BaseExercise class which returns an integer valua representing minutes from the exercise's module's closing time 'til a caller provided datetime object.

### CODE REVIEW

File: locale/fi/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -2615,13 +2627,33 @@ msgstr ""
...
+#| "\t\t\tFor this service also API token is tranferred,\n"
+#| "\t\t\tthus giving this service access to A+ API with your privileges.\n"
+#| "\t\t\tThat means service can for example read your student number,\n"
+#| "\t\t\tcourse progress and submit solutions to exercises.\n"
+#| "\t\t\tCourse staff has enabled this feature and have checked that\n"
+#| "\t\t\tthe service behaves with your best interest in mind.\n"
+#| "\t\t\t"
 msgid ""
 "\n"
 "\t\t\tFor this service also API token is tranferred,\n"
```
Comment (raphendyr) [inline locale/fi/LC_MESSAGES/django.po]: `tranferred` damn. I have created typo here :(

File: locale/fi/LC_MESSAGES/django.po [hunk 2]
```diff
@@ -2615,13 +2627,33 @@ msgstr ""
 "ylläpitäjälle."
 
 #: external_services/templates/external_services/lti_form.html:23
+#, fuzzy
```
Comment (raphendyr) [inline locale/fi/LC_MESSAGES/django.po]: You can manually remove this fuzzy block.

**Comment by raphendyr** [comment]
looks good. If you can check the last few comments then it's ready for merging.

**Comment by piehei** [comment]
Done'd

**Pull request merged**